Core/clim-core/panes/layout-protocol.lisp

```diff
@@ -466,20 +454,35 @@ which changed during the current execution of CHANGING-SPACE-REQUIREMENTS.
                                               &rest space-req-keys
                                               &key resize-frame &allow-other-keys)
   (declare (ignore resize-frame space-req-keys))
-  ;; Clear current width and height.
-  (setf (pane-space-requirement pane) nil
-        (pane-current-width pane) nil
-        (pane-current-height pane) nil))
+  ;; Clear the cached value
+  (setf (pane-space-requirement pane) nil))
 
 (defmethod change-space-requirements ((pane layout-protocol-mixin)
                                       &key resize-frame &allow-other-keys)
-  (when-let ((parent (sheet-parent pane)))
-    (change-space-requirements parent :resize-frame resize-frame)))
+  ;; do nothing here
+  nil)
 
 (defmethod change-space-requirements :after ((pane layout-protocol-mixin)
                                              &key resize-frame &allow-other-keys)
-  (declare (ignore resize-frame))
-  (note-space-requirements-changed (sheet-parent pane) pane))
+  (when-let ((parent (sheet-parent pane)))
+    (if resize-frame
+        ;; From Spec 29.3.4: "If resize-frame is true, then
+        ;; layout-frame will be invoked on the frame". Here instead of
+        ;; call directly LAYOUT-FRAME, we call
+        ;; CHANGE-SPACE-REQUIREMENTS on the parent and it travels all
+        ;; the way up to the top-level-sheet-pane which then invokes
+        ;; the layout protocol calling LAYOUT-FRAME. The rationale of
+        ;; this is:
+        ;; 1. we can't call (LAYOUT-FRAME (PANE-FRAME pane)) on a
+        ;;   menu because with the actual implementation of menu it
+        ;;   will layout the main application and not the menu frame.
+        ;; 2. we automatically clear the cached values of
+        ;;    space-requirements for the involved panes.
+        ;; -- admich 2020-08-11
+        (if (top-level-sheet-pane-p pane)
```

Comment:
this function is forward-referenced from the file composition.lisp (it is defined after the first use)

Reply:
Yes, it's true. I don't have a solution, moving here the definition doesn't seem right. 

Maybe the right things it is to change the menu implementation before this PR and then use what I have written before:
```
(defmethod change-space-requirements :after ((pane layout-protocol-mixin)
                                             &key resize-frame &allow-other-keys)
  (when-let ((parent (sheet-parent pane)))
    (if resize-frame
        (if *changing-space-requirements*
            (unless (find pane *changed-space-requirements* :key #'second)
              (push (list (pane-frame pane) pane resize-frame)
                    *changed-space-requirements*))
            (layout-frame (pane-frame pane)))
        (note-space-requirements-changed parent pane))))
```